### PR TITLE
dedicated docs for YAML metadata

### PR DESCRIPTION
**Pull request by bendichter**
* add dedicated docs page for yaml
* expand schema for NWBFile

## Checklist

- [x] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [x] Have you ensured the PR description clearly describes the problem and solutions?
- [x] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [x] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [x] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?

### CODE REVIEW

**Comment by bendichter** [comment]
This was motivated by Angelique's question. Hopefully this will be a step forward.

File: docs/user_guide/yaml.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    metadata = spikeglx_interface.get_metadata()
+    metadata = dict_deep_update(metadata, metadata_from_yaml)
+
+    spikeglx_interface.run_conversion(
+        save_path="path/to/destination.nwb",
+        metadata=metadata
+    )
+
+Note that any metadata extracted in by ``spikeglx_interface.get_metadata()`` will be overwritten by the YAML data.
```
Comment (weiglszonja) [inline docs/user_guide/yaml.rst:80]: I would go with "update" instead of overwritten.

File: docs/user_guide/yaml.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+The above YAML is common to all :py:class:`.BaseDataInterface`, :py:class:`~neuroconv.nwbconverter.NWBConverter`,
+or :py:class:`~neuroconv.nwbconverter.ConverterPipe`, and an analogous workflow for incorporating this data will work
+for each. Specific interfaces and converter will have additional fields, which you can see using the method
+:func:`DataInterface.get_metadata_schema() <neuroconv.basedatainterface.BaseDataInterface.get_metadata_schema>` or
+:func:`NWBConverter.get_metadata_schema() <neuroconv.nwbconverter.NWBConverter.get_metadata_schema>` or
```
Comment (weiglszonja) [inline docs/user_guide/yaml.rst:86]: Unfinished sentence? Also maybe put a reference here to https://neuroconv.readthedocs.io/en/main/user_guide/schemas.html#metadata-schema-and-data for example how the schema could look like.

**Comment by CodyCBakerPhD** [comment]
Minor suggestions and trying to get API docs for the base interface working: https://github.com/catalystneuro/neuroconv/pull/242

Otherwise LGTM

File: docs/user_guide/yaml.rst (hunk 1 above)
Comment (bendichter) [inline docs/user_guide/yaml.rst:80]:
```suggestion
Note that any metadata extracted in by ``spikeglx_interface.get_metadata()`` will be updated by the YAML data.
```

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**